vxquery-core/src/main/java/org/apache/vxquery/xmlquery/translator/XMLQueryTranslator.java

```diff
@@ -1503,8 +1521,10 @@ private LogicalVariable translatePathExpr(PathExprNode pe, TranslationContext tC
                         tCtx.op = select;
                         ctxExpr = vre(tCtx.varScope.lookupVariable(XMLQueryCompilerConstants.DOT_VAR_NAME)
                                 .getLogicalVariable());
-                        tCtx.popVariableScope();
-                        tCtx = tCtx.popContext();
+                        if (!popScope) {
+                            tCtx.popVariableScope();
```

Comment:
This if statement should be removed. I will have code that actually makes this work in the near future.